draft-ietf-privacypass-key-consistency.md

```diff
@@ -191,94 +229,68 @@ below.
 |          |
 +----------+
 ~~~
-{: #fig-disc-proxy title="Single Proxy Discovery Example"}
-
-If this proxy is trusted, then all users which request a key from this server are assured they have
-a consistent view of the server key. However, if this proxy is not trusted, operational risks may arise:
-
-- The proxy can collude with the server to give per-user keys to clients.
-- The proxy can give all users a key owned by the proxy, and either collude with the server to use this
-  key or retroactively use this key to compromise user privacy when users later make use of the key.
-
-Mitigating these risks can be done in a variety of ways. For example, clients may demand tamper-proof
-proof evidence that the key is consistent and correct for the server, using techniques described in {{server-based}}.
-Clients may gossip amongst themselves to determine if they are being served different keys.
-Alternatively, the clients may attempt to confirm the key provided by the proxy, as described in {{shared-proxy-with-confirmation}}.
+{: #fig-disc-proxy title="Shared Cache Discovery Example"}
+
+The validity window of the cache's response can impact the overall consistency guarantees.
+In particular, a system needs to ensure that a server cannot rotate its keys too often in order
+to divide clients into smaller groups based on when keys are acquired. Such considerations are
+already highlighted within the Privacy Pass ecosystem, more discussion can be found in {{PRIVACY-PASS-ARCH}}.
+Setting a minimum validity period limits the ability of a server to rotate keys, but also
+limits the rate of key rotation.
+
+Querying a cache for its stored copy of a key leaks information to that cache.
+There are several mitigations for this leak. For example, clients could obtain the
+contents of a cache and query it locally. Alternatively, clients could remotely query
+the cache using privacy-preserving queries (e.g., a private information retrieval (PIR)
+protocol). In the case where the cache is downloaded locally, it should be considered
+stale and re-fetched periodically. The frequency of such updates can likely be infrequent
+in practice, as frequent key updates or rotations may affect privacy. Downloading the
+entire cache works best if there are a small number of entries, as it does not otherwise
+impose bandwidth costs on each client that may be impractical.
 
-## Shared Proxy with Key Confirmation {#shared-proxy-with-confirmation}
+If this cache is trusted, then all clients which request a key from this server are
```

Comment:
"is trusted" is too direct and absolute.  Clients don't need to unconditionally trust a cache, they only have to have some expectation that it not collude with the provider of the key/configuration to unmask clients.  That is, the cache is entrusted with the responsibility not to take deliberate action that would compromise client security.  It helps a little that operating honestly is cheaper than otherwise, but this trust element still exists.